### PR TITLE
remove is_bootstrap parameter

### PR DESCRIPTION
**Pull request by chenqin**
remove is_bootstrap parameter from rabit api 
implicit use signature and loadcheckpoint flag instead

### CODE REVIEW

**Comment by chenqin** [comment]
@hcho3 @trivialfis @CodingCat this is minor pr remove is_bootstrap parameter from rabit api to make it more user friendly.  https://github.com/dmlc/xgboost/pull/4808 would benefit by having a smaller code change.

File: src/allreduce_robust.h [hunk 1]
```diff
@@ -643,6 +639,8 @@ o   *  the input state must exactly one saved state(local state of current node)
   std::string local_chkpt[2];
   // version of local checkpoint can be 1 or 0
   int local_chkpt_version;
+  // if checkpoint were loaded, used to split results boostrap cache with seqno cache
```
Comment (CodingCat) [inline src/allreduce_robust.h]: split -> distinguish?

Comment (chenqin) [reply src/allreduce_robust.h]: fixed.

**Comment by chenqin** [comment]
> could you complete the description about when we load bootstrap cache now?

goog catch, let me update documentation as well.

**Comment by chenqin** [comment]
@CodingCat done

**Pull request merged**